### PR TITLE
Fix and Enhance ExprDisplayViewRange

### PR DESCRIPTION
**Pull request by Fusezion**
**EDIT**
fixes and enhances the ExprDisplayViewRange class to support add/remove/reset changers
\- Shane

Related Issue: #469

### CODE REVIEW

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+@Examples({"set {_items::*} to namespacedkey of every sword",
+		"loop {_items::*}:",
+		"\tbroadcast loop-value",
+		"\tbroadcast namespace of loop-value",
+		"\tbroadcast namespace key of loop-value"})
+@Since("INSERT VERSION")
+public class ExprNamespaceKeyOf extends SimplePropertyExpression<NamespacedKey, String> {
+
+	static {
+		register(ExprNamespaceKeyOf.class, String.class, "namespace [:key]", "namespacedkeys");
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java:25]: I feel like this should be:
```suggestion
		register(ExprNamespaceKeyOf.class, String.class, "(namespace|:key)", "namespacedkeys");
```

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+@Name("NamespacedKey - Namespace/Key")
+@Description({"Gets the namespace or namespace key of a NamespacedKey object."})
+@Examples({"set {_items::*} to namespacedkey of every sword",
+		"loop {_items::*}:",
+		"\tbroadcast loop-value",
+		"\tbroadcast namespace of loop-value",
+		"\tbroadcast namespace key of loop-value"})
+@Since("INSERT VERSION")
+public class ExprNamespaceKeyOf extends SimplePropertyExpression<NamespacedKey, String> {
+
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java:23]: not sure where to put this, so ill put this at the top
I noticed this class is using tabs not spaces

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	static {
+		register(ExprNamespaceKeyOf.class, String.class, "namespace [:key]", "namespacedkeys");
+	}
+
+	private boolean key;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		key = parseResult.hasTag("key");
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java:32]: just for consistency sake with how ive been doing things
```suggestion
		this.key = parseResult.hasTag("key");
```

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	@Nullable
+	public String convert(NamespacedKey namespacedKey) {
+		return key ? namespacedKey.getKey() : namespacedKey.getNamespace();
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java:39]: same as above
```suggestion
		return this.key ? namespacedKey.getKey() : namespacedKey.getNamespace();
```

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	@Override
+	public Class<? extends String> getReturnType() {
+		return String.class;
+	}
+
+	@Override
+	protected String getPropertyName() {
+		return "namespace" + (key ? " key" : "");
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java:49]:
```suggestion
		return this.key ? "key" : "namespace";
```

**Review by ShaneBeee — CHANGES_REQUESTED** [review]
just a few changes

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java (hunk 1 above)
Comment (Fusezion) [reply src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java:25]: I didn't do that due to `key of %namespacedkeys%`, this was the same issue we had with lockable keys

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java (hunk 2 above)
Comment (Fusezion) [reply src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java:23]: ah whoops sorry, my current project uses tabs forgot to swap back for this

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java (hunk 1 above)
Comment (ShaneBeee) [reply src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java:25]: I don't know where that is used... that said, in my honest opinion, I don't think this expression is really needed.
Due to the fact that:
a) I can't see many people using it at all
b) you can easily stringily and split

Comment (Fusezion) [reply src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprNamespaceKeyOf.java]: It's not used anywhere but it's just in the case of a conflict between anything with `key` being the only thing that's gonna be used for matching *aside from literals*

**Comment by Fusezion** [comment]
Gonna go ahead and just remove the namespace stuff, all that'll remain is the 2 commits you did for ExprDisplayViewRange.

**Pull request merged**